packages/react/src/transactions/adapters/AbstractContractCallGateway.ts

```diff
@@ -41,21 +48,29 @@ export abstract class AbstractContractCallGateway<TRequest extends AnyTransactio
 
     const { contractAddress, encodedData } = await this.createEncodedData(request);
 
+    // skip gas estimation if debug mode is enabled
+    if (this.logger.level === LoggerLevel.Debug) {
```

Comment:
I would rather pass `debug` flag here rather than a second hand information (i.e. the log level based on `debug` value), WDTY?

Reply:
the thing is that you need to pass this extra param all the way down through the class hierarchy, and it is used only at the bottom, is there any better way to do it?

Reply:
I understand, but this is no different than the effort already done here. Instead of passing the `logger` you would pass the `config` (or `config.debug` flag).

Reply:
ok, I changed to pass down `config` instead of `logger`